### ISSUE TITLE
Parsing regression (?)

### ISSUE THREAD

**Issue by ISSOtm** [rednex/rgbds#289]
https://github.com/zzazzdzz/fools2018 compiles correctly with RGBDS 0.3.3, but errors out with 0.3.7:
```
ERROR: main.asm(7) -> main_sram2.asm(6) -> engine/text_engine.asm(369) -> includes/charset.asm(499):
    syntax error
error: Assembly aborted in pass 1 (1 errors)!
```
(Although the setup surrounding RGBDS is kinda hacky, this invocation should be completely vanilla.)

**Comment by Ben10do** [comment]
Am I right in saying that that error corresponds to these lines?

```
496     db $ff,$05
497 
498 Char38: ; "
499     db $00,$01,$02
450     db $20,$21,$22
```

It could be getting confused by the `"` in the comment on the line before, though I've no idea what parsing changes could've introduced this.

**Comment by Ben10do** [comment]
Would it be easy for you to find out which rgbds version is the first that fails to compile?

**Comment by AntonioND** [comment]
It's probably because of https://github.com/rednex/rgbds/commit/efdd42c6a80945c78b396054ebfdbd4fd9f1f1a6

**Comment by ISSOtm** [comment]
@Ben10do Well, I tried a minimal repro with the same intuition, and it didn't work either. Not sure what it could be besides the quote, though?
Also, I could try compiling it with other versions, if needed.

**Comment by AntonioND** [comment]
Have you tried git bisect to see where the problem started?

**Comment by ISSOtm** [comment]
Well, since it's not my repo, I didn't try that :stuck_out_tongue:

**Comment by AntonioND** [comment]
Ok, I thought that maybe you had done more tests than just checking 0.3.3 and 0.3.7. Do you have a minimal example that triggers the bug? @Ben10do, could you post what you used?

**Comment by Ben10do** [comment]
@AntonioND This is the quick-and-dirty test that I tried, but it compiles without error:

```
SECTION "x", ROM0

Test: ; "
    db 0, $1F
```

**Comment by ISSOtm** [comment]
That is what I tried as well. I then wondered if it could be caused by an INCLUDE or smth, but I couldn't compile the file alone (missing macros, IIRC). I didn't look any further, though.

**Comment by ISSOtm** [comment]
Bumping this - @DevEd2 got a similar error with RGBDS 0.3.7, declared on the line after a `ld [ScrollPos+1], a ; "`. Apparently this is an actual bug, but why can't it be tripped with a minimal example?

**Comment by DevEd2** [comment]
This appears to reproduce the bug (also gives a spurious "expression must be 8-bit"):
```section "test",rom0
test:
	db "hello, world"
	db 123
test2:
	db "test" ; "
	db "test2"
.test3
	db	"asdasdasd" ;"
	dw	456```

**Comment by AntonioND** [comment]
Maybe the double quotes are, not the label?

**Comment by ISSOtm** [comment]
As noted above by both I and Ben10do, the double quotes are not enough to trigger the error. Removing the last two lines seems to make the thing compile fine?

**Comment by AntonioND** [comment]
I can't reproduce the error with that example (``rgbasm -o bug.o bug.asm``).

**Comment by ISSOtm** [comment]
Huh, same. @DevEd2?

**Comment by dbrotz** [comment]
I can't reproduce this with the case linked in the first message.

@ISSOtm Are you compiling rgbds yourself or using a prebuilt version?

**Comment by ISSOtm** [comment]
I'm using it on Linux, thus I have to build it myself

**Comment by dbrotz** [comment]
I'm also testing it on Linux, but I don't get the error when I do the following.

```
cd fools2018/savefile
rgbasm -o main.o main.asm
```

It assembles without any errors. Is there anything special you have to do?

**Comment by dbrotz** [comment]
rgbasm v0.3.7-34-g5cb6c4a

**Comment by ISSOtm** [comment]
Well, maybe it has been fixed since? I'm using `rgbasm v0.3.7`.

**Comment by NieDzejkob** [comment]
Might be related to #326

**Comment by dbrotz** [comment]
I agree that this was caused by #326. The confusion over reproducing this makes sense given that it only happens when you use Windows line endings. It should be fixed now that #370 has been merged.

**Comment by ISSOtm** [comment]
The following does not generate an error, I'll assume this is fixed then.

```
issotm@link-panda tmp.jXWovMS7IM% printf 'SECTION "test",ROM0\r\ntest:\r\n\tdb 1 ; "\r\n.test2\r\n\tdb 2 ; "' > test.asm
issotm@link-panda tmp.jXWovMS7IM% cat test.asm
SECTION "test",ROM0
test:
	db 1 ; "
.test2
	db 2 ; "
issotm@link-panda tmp.jXWovMS7IM% rgbasm test.asm
issotm@link-panda tmp.jXWovMS7IM%
```
